### PR TITLE
feat: implement cross-tag task movement functionality

### PR DESCRIPTION
**Pull request by mm-parthy**
Implements the ability to move tasks and subtasks between different tag contexts. This feature allows users to reorganize tasks across different project contexts (branches, features, experiments) while maintaining task integrity and proper validation.

Key features:
- Move tasks between different tags while preserving task structure
- Comprehensive validation to prevent data loss and conflicts
- Enhanced error handling with user-friendly messages
- Support for both CLI and MCP interfaces
- Extensive test coverage for cross-tag scenarios

## Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Documentation update

## Testing

- [x] I have tested this locally
- [x] All existing tests pass
- [x] I have added tests for new functionality

## Changeset

- [ ] I have created a changeset (or this change doesn't need one)

## Additional Notes

This PR adds significant functionality to the tagged task system, enabling more flexible task organization across different development contexts. The implementation includes:

- New `move-task-cross-tag` MCP tool and CLI command
- Enhanced validation logic to prevent invalid cross-tag moves
- Comprehensive error handling and user feedback
- Unit tests covering various cross-tag movement scenarios
- Documentation updates explaining the new functionality

The feature maintains backward compatibility and follows existing patterns in the codebase.

## Summary by CodeRabbit

- New Features
  - Cross‑tag task moves via CLI (single or multiple IDs) with --from-tag/--to-tag and dependency controls (--with-dependencies, --ignore-dependencies).
  - Batch move summaries, automatic target tag creation, and fallback to current tag.

- Documentation
  - Comprehensive cross‑tag movement guide and README examples with usage, flags, and troubleshooting.

- UI/Help
  - Improved error messages, resolution options, and contextual hints for dependency and subtask move issues.

- Tests
  - Extensive unit and integration tests covering cross‑tag scenarios, dependencies, subtasks, and edge cases.

- Bug Fixes
  - Fixed add‑tag --from-branch runtime error and more robust subtask ID handling.

### CODE REVIEW

**Comment by Crunchyman-ralph** [comment]
### ⚠️ Potential Silent Mode Leak

In `mcp-server/src/core/direct-functions/move-task-cross-tag.js` lines 94-131:

The silent mode enable/disable pattern could leak if an error occurs between enabling and disabling. Consider using a try/finally block:

```javascript
try {
    enableSilentMode();
    // ... your logic here
    const result = await moveTasksBetweenTags(...);
    return { success: true, data: result };
} catch (error) {
    // handle error
} finally {
    disableSilentMode();
}
```

This ensures silent mode is always restored even if an error occurs.

**Comment by Crunchyman-ralph** [comment]
### 🔍 ID Type Consistency Issue

In `scripts/modules/dependency-manager.js`, there's potential for ID comparison issues with mixed string/number types:

```javascript
const hasDependency = task.dependencies.some(depId => 
  sourceIds.includes(depId)
);
```

Consider normalizing IDs before comparison:

```javascript
const normalizedSourceIds = sourceIds.map(id => String(id));
const hasDependency = task.dependencies.some(depId => 
  normalizedSourceIds.includes(String(depId))
);
```

This prevents issues where `'1' \!== 1` could cause missed dependencies.

**Comment by Crunchyman-ralph** [comment]
### 💡 Refactoring Suggestion: Function Complexity

The `moveTasksBetweenTags` function in `scripts/modules/task-manager/move-task.js` is quite long (~200+ lines). Consider breaking it down into smaller, focused functions:

```javascript
// Suggested refactor structure:
async function moveTasksBetweenTags(tasksPath, sourceIds, sourceTag, targetTag, options, context) {
    // 1. Validation phase
    const validation = await validateMove(tasksPath, sourceIds, sourceTag, targetTag);
    
    // 2. Load and prepare data
    const { rawData, sourceTasks, allTasks } = await prepareTaskData(validation);
    
    // 3. Handle dependencies
    const tasksToMove = await resolveDependencies(sourceTasks, allTasks, options);
    
    // 4. Execute move
    const result = await executeMoveOperation(tasksToMove, sourceTag, targetTag, rawData);
    
    // 5. Save and return
    return await finalizeMove(result, tasksPath);
}
```

This would improve readability, testability, and maintainability.

**Review by Crunchyman-ralph — APPROVED** [review]
## ✅ Approved with Minor Suggestions

Great work on implementing the cross-tag task movement feature! This is a well-thought-out addition to Task Master.

### Summary
- **Feature completeness**: ✅ All requirements met
- **Code quality**: ✅ Good structure and organization  
- **Testing**: ✅ Comprehensive test coverage
- **Documentation**: ✅ Excellent documentation and examples
- **Error handling**: ✅ Robust with helpful user messages

### Required Fix
Please address the **silent mode leak issue** before merging - using try/finally blocks will ensure proper cleanup.

### Nice-to-Have Improvements (can be follow-up PRs)
1. ID type normalization for consistency
2. Function decomposition for better maintainability
3. Consider adding transaction/rollback support for multi-task operations
4. Progress indicators for large batch operations

The core functionality is solid and the feature will provide significant value to users. Once the silent mode issue is fixed, this is ready to merge!

🎉 Excellent work on this feature implementation!

**Comment by mm-parthy** [comment]
@coderabbitai review

File: tests/unit/scripts/modules/commands/README.md
Comment (Crunchyman-ralph) [inline tests/unit/scripts/modules/commands/README.md:1]: should be removed

File: tests/unit/scripts/modules/task-manager/complexity-report-tag-isolation.test.js [hunk 1]
```diff
@@ -90,6 +90,7 @@ jest.unstable_mockModule('../../../../../scripts/modules/utils.js', () => ({
 		}
 		return path.join(projectRoot || '.', basePath);
 	}),
+	traverseDependencies: jest.fn((sourceTasks, allTasks, options = {}) => []),
```
[the remark this answers is not in the document]
Comment (Crunchyman-ralph) [reply tests/unit/scripts/modules/task-manager/complexity-report-tag-isolation.test.js:93]: good point

File: tests/unit/scripts/modules/commands/README.md
Comment (Crunchyman-ralph) [reply tests/unit/scripts/modules/commands/README.md:1]: keeping for prosperity and to aim for other unit tests to look like that.
